provider/pkg/provider/remote/copy.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"github.com/pulumi/pulumi-go-provider/infer"
+	"github.com/pulumi/pulumi-go-provider/infer/types"
+)
+
+type Copy struct{}
+
+var _ = (infer.Annotated)((*Copy)(nil))
+
+// Copy implements Annotate which allows you to attach descriptions to the Copy resource.
+func (c *Copy) Annotate(a infer.Annotator) {
+	a.Describe(&c, "Copy an Asset or Archive to a remote host.")
+}
+
+type CopyInputs struct {
+	Connection *Connection          `pulumi:"connection" provider:"secret"`
+	Triggers   *[]interface{}       `pulumi:"triggers,optional" provider:"replaceOnChanges"`
+	Source     types.AssetOrArchive `pulumi:"source"`
+	RemotePath string               `pulumi:"remotePath"`
+}
+
+func (c *CopyInputs) Annotate(a infer.Annotator) {
+	a.Describe(&c.Connection, "The parameters with which to connect to the remote host.")
+	a.Describe(&c.Triggers, "Trigger replacements on changes to this input.")
+	a.Describe(&c.Source, "An asset or an archive to upload as the source of the copy. It must be path based.")
```

Comment:
What does "it must be path based" mean here? Does this fail if someone passes in a StringAsset and sets `RemotePath` to "~/test.txt"?

Reply:
Might also be nice to include specific examples of specific asset and archive types and how they'll be transferred.
- Does an AssetArchive get unpacked at the other end or does it depend on the remotePath? If it's remote path is `~/dest.tar.gz` does it keep it packed?
- Does an Asset assume the remote path is a filename and an archive assume it's a directory?